### PR TITLE
mtest: Lower default max buffer size

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

The previous value was negative when MPI_Aint is a 32-bit signed
integer. Decrement it by one to ensure a positive buffer size.

## Expected Performance Changes

none

## Known Issues

none

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Passes tests (included warning check)
* [x] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [x] Commits are self-contained and do not do two things at once
* [x] Remove xfail from the test suite when fixing a test
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
A negative max buffer size led to this conditional never being satisfied https://github.com/pmodels/mpich/blob/master/test/mpi/dtpools/src/dtpools.c#L151. This resulted in an infinite loop in `DTP_obj_create`.

**Comment by hzhou** [comment]
test:jenkins/ch3/tcp

@raffenet I added a sanity check to prevent such case from entering a dead loop.

**Comment by raffenet** [comment]
@hzhou thanks. I have another concern that the default max buffer setting results in Jenkins failures on 32-bit systems that are hidden by our envvar setting in the build scripts. I will open a separate issue for that.

**Comment by hzhou** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
New version sets the default to 1GB.
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
macOS failure is not related to dtpools, so I think it can be ignored. This should be good to go.

File: test/mpi/util/mtest_common.c [hunk 1]
```diff
@@ -18,7 +18,7 @@
 */
 MPI_Aint MTestDefaultMaxBufferSize()
 {
-    MPI_Aint max_size = 2147483648;
+    MPI_Aint max_size = 1073741824;
```
Comment (hzhou) [inline test/mpi/util/mtest_common.c]: It is odd to have an arbitrarily looking constant in the code. Can we either round it to `1000000000` or use hex form `0x40000000`? It is just a nitpick. I am approving the review regardless of the change.

**Pull request merged**